crates/sovereign-sdk/rollup-interface/src/state_machine/zk/mod.rs

```diff
@@ -25,6 +27,60 @@ pub mod light_client_proof;
 /// The ZK proof generated by the [`ZkvmHost::run`] method.
 pub type Proof = Vec<u8>;
 
+#[cfg(feature = "native")]
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+/// Information about a local prover's execution.
+pub struct LocalInfo {
+    /// Segments count
+    pub segments: usize,
+    /// Total cycles count
+    pub total_cycles: u64,
+    /// User cycles count
+    pub user_cycles: u64,
+    /// Paging cycles count
+    pub paging_cycles: u64,
+    /// Reserved cycles count
+    pub reserved_cycles: u64,
+}
+
+#[cfg(feature = "native")]
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+/// Information about a bonsai prover's execution.
+pub struct BonsaiInfo {
+    /// Session ID of the proof request
+    pub session_id: String,
+    /// Count of segments in this proof request
+    pub segments: usize,
+    /// Total cycles run within guest
+    pub total_cycles: u64,
+    /// User cycles run within guest, slightly below total overhead cycles
+    pub user_cycles: u64,
+    /// Type of the proof receipt
+    pub receipt_type: ReceiptType,
+}
+
+#[cfg(feature = "native")]
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+/// Information about a boundless prover's execution.
+pub struct BoundlessInfo {
+    /// Request ID of the proof request
+    pub request_id: String,
+    /// Mcycles count
+    pub mcycles_count: u64,
+}
+
+#[cfg(feature = "native")]
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+/// Information about prover's execution.
+pub enum ProvingInfo {
+    /// Local prover info
+    Local(LocalInfo),
+    /// Bonsai prover info
+    Bonsai(BonsaiInfo),
+    /// Boundless prover info
+    Boundless(BoundlessInfo),
+}
```

Comment:
because it represents one session info, feel like should be like this:

```suggestion
pub enum ProvingSessionInfo {
    /// Local prover proving session info
    Local(LocalProverProvingSessionInfo),
    /// Bonsai prover proving session info
    Bonsai(BonsaiProverProvingSessionInfo),
    /// Boundless prover proving session info
    Boundless(BoundlessProverProvingSessionInfo),
}
```

Reply:
all done, thx for the suggestion🙏🏻